### PR TITLE
Issue 56

### PR DESCRIPTION
**Pull request by jsstevenson**

### CODE REVIEW

File: analysis/graph/db_helper.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+"""Utility function to load/reload graph for development."""
+from metakb.database import Graph
+from metakb import PROJECT_ROOT
+import json
+
+
+g = Graph(uri="bolt://localhost:7687", credentials=("neo4j", "admin"))
+g.clear()
+
+fpath = PROJECT_ROOT / 'analysis' / 'graph' / 'civic_cdm_v3.json'
```
[the remark this answers is not in the document]
Comment (jsstevenson) [reply analysis/graph/db_helper.py:10]: Yeah -- issue-59 deprecates this script but I'll update in the meantime

File: metakb/database.py [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+                descriptor[key].append(expression['value'])
+            else:
+                descriptor[key] = [expression['value']]
+        nonnull_keys = [f"{key}:${key}"
+                        for key in ('id', 'label', 'description', 'xrefs',
+                                    'alternate_labels', 'structural_type',
+                                    'expressions_transcript',
+                                    'expressions_genomic',
+                                    'expressions_protein',
+                                    'ref_allele_seq')
```
[the remark this answers is not in the document]
Comment (jsstevenson) [reply metakb/database.py:186]: oops missed molecule_context

File: analysis/graph/db_helper.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+"""Utility function to load/reload graph for development."""
+from metakb.database import Graph
+from metakb import PROJECT_ROOT
+import json
+
+
+g = Graph(uri="bolt://localhost:7687", credentials=("neo4j", "admin"))
+g.clear()
+
+fpath = PROJECT_ROOT / 'analysis' / 'graph' / 'civic_cdm_latest.json'
```
Comment (korikuzma) [inline analysis/graph/db_helper.py:10]: If we keep this, wondering if we could just have it point to the file in `/data/tranform/civic_cdm.json`

File: metakb/database.py [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+                                             'expressions_protein',
+                                             'ref_allele_seq'))]
+
+        # handle extensions
+        variant_groups = None
+        extensions = descriptor.get('extensions')
+        if extensions:
+            for ext in extensions:
+                name = ext['name']
+                if name == 'variant_groups':
```
Comment (korikuzma) [inline metakb/database.py:212]: This is now `variant_group`: https://github.com/cancervariants/metakb/blob/issue-56/metakb/transform/civic.py#L502

File: metakb/database.py [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+                                  f"Query: {query}\nDescriptor: {descriptor}")
+                    raise exception
+
+    @staticmethod
+    def _add_proposition(tx, proposition: Dict):
+        """Add Proposition object to DB.
+        :param Dict proposition: must include `disease_context`, `therapy`,
+            and `has_originating_context` fields.
+        """
+        proposition['id'] = proposition['id']
```
Comment (korikuzma) [inline metakb/database.py:268]: Is this needed still?

**Review by korikuzma — APPROVED** [review]

File: analysis/graph/db_helper.py (hunk 3 above)
Comment (jsstevenson) [reply analysis/graph/db_helper.py:10]: Maybe we could just fold this functionality into the `cli` module and grab from the respective `/data/transform` folders -- I'll add to issue-59

**Review by MayLiu27 — APPROVED** [review]
Fantastic! A README is definitely needed. Program works perfect on my local machine. Great work!
One thing is, the `db_helper.py ` loads one source of cdm.json for now, maybe it's in the todo list, but should we consider a way to sequentially load all sources of cdm.json from `data` folder?

**Comment by jsstevenson** [comment]
Thanks @MayLiu27! 

I spent some time last week updating the README for the workflow as of this branch: https://github.com/cancervariants/metakb/blob/issue-56/README.md

I also have a PR (currently in draft) that includes more robust commands to run everything end-to-end for all included resources. We may do some more small fiddling on it but it should be mostly final, if you want to take a look: https://github.com/cancervariants/metakb/pull/63

**Comment by korikuzma** [comment]
These changes are already merged into master. Can we close this? @jsstevenson

**Pull request closed without merging**